src/saveload/details.rs

```diff
@@ -0,0 +1,176 @@
+use std::error::Error;
+
+use serde::de::DeserializeOwned;
+use serde::ser::Serialize;
+
+use {Component, Entity, ReadStorage, SystemData, WriteStorage};
+
+use super::marker::Marker;
+use error::NoError;
+
+#[derive(Serialize, Deserialize)]
+#[serde(bound = "")]
+pub struct EntityData<M: Marker, E, T: Components<M::Identifier, E>> {
+    pub marker: M,
+    pub components: T::Data,
+}
+
+/// Trivial component is anything that is Component + Copy + DeserializeOwned + Serialize
+pub trait SerializableComponent: Component + DeserializeOwned + Serialize {}
+impl<C> SerializableComponent for C
+where
+    C: Component + DeserializeOwned + Serialize,
+{
+}
+
+/// This trait should be implemented in order to allow component
+/// to be serializeble with `SerializeSystem`.
+/// It is automatically implemented for all `SerializableComponent`s
+pub trait SaveLoadComponent<M>: Component {
```

Comment:
probably needs a rename: save/load is a game term, too specific for specs ECS (although, I'm not 100% confident about this)

Reply:
I can rename it after someone make a better suggestion 😄

Reply:
Serialize/Deserialize ?